packages/data-api/src/AnalyticsFetchQuery.js

```diff
@@ -0,0 +1,202 @@
+    }
+    const { groupByPeriodField, sum, getLatestPerPeriod } = this.aggregation.switches;
+    const fields = [...this.getCommonFields()];
+
+    if (groupByPeriodField) {
+      fields.push(groupByPeriodField);
+    }
+
+    if (sum) {
+      fields.push('SUM(value::NUMERIC)::text as value');
+      fields.push('MAX(type) as type');
+      if (!getLatestPerPeriod) {
+        fields.push('MAX(date) as date');
+      }
+    }
+    return `SELECT ${fields.join(', ')}`;
+  }
+
+  getA1InnerJoins() {
+    const joins = [
+      'INNER JOIN entity_codes_and_relations ON entity_codes_and_relations.code = analytics.entity_code',
+      this.createInnerJoin(this.dataElementCodes, 'data_element_code'),
+    ];
+    return joins.join('\n');
+  }
+
+  getPeriodConditionsAndParams() {
+    const periodConditions = [];
+    const periodParams = [];
+    if (this.startDate) {
+      periodConditions.push('date >= ?');
+      periodParams.push(this.startDate);
+    }
+    if (this.endDate) {
+      periodConditions.push('date <= ?');
+      periodParams.push(this.endDate);
+    }
+    return { periodConditions, periodParams };
+  }
+
+  getA1WhereClause() {
+    const { periodConditions, periodParams } = this.getPeriodConditionsAndParams();
+
+    if (periodConditions.length === 0) return '';
+
+    this.paramsArray.push(...periodParams);
+    return `WHERE ${periodConditions.join(' AND ')}`;
+  }
+
+  getA1GroupByClause = () => {
+    if (!this.isAggregating) return '';
+
+    const groupByFields = [...this.getCommonFields()];
+    const { groupByPeriodField } = this.aggregation.switches;
+    if (groupByPeriodField) groupByFields.push(groupByPeriodField);
+    return `GROUP BY ${groupByFields.join(', ')}`;
+  };
+
+  getLatestPerPeriodClause() {
+    if (!this.isAggregating || !this.aggregation.switches.getLatestPerPeriod) return '';
+
+    // add where condition for each non-calculated table selected in a1
+    const joinFields = [...this.getCommonFields()];
+    const { groupByPeriodField } = this.aggregation.switches;
+    if (groupByPeriodField) joinFields.push(groupByPeriodField);
+    const joinConditions = joinFields.map(field => `${field} = a1.${field}`);
+
+    // add start/end date clauses to limit results and speed things up
+    const { periodConditions, periodParams } = this.getPeriodConditionsAndParams();
+    this.paramsArray.push(...periodParams);
+
+    const conditions = [...joinConditions, ...periodConditions];
+
+    const fields = this.aggregation.switches.sum
+      ? ANSWER_SPECIFIC_FIELDS.filter(field => field !== 'value')
+      : ANSWER_SPECIFIC_FIELDS;
+
+    return `
+      CROSS JOIN LATERAL (
+        SELECT ${fields}
+        FROM analytics
+        INNER JOIN entity_codes_and_relations ON entity_codes_and_relations.code = analytics.entity_code
+        WHERE ${conditions.join('\n      AND ')}
+        ORDER BY date DESC
+        LIMIT 1
+      ) as latestPerPeriod
```

Comment:
Small thing, but I really appreciate you giving this join table a human readable name! I just did a loggable query on your branch and found it a lot more understandable than it used to be! 🎉